src/utils/util/resolver-pipeline/oscal-profile-resolve-merge.xsl

```diff
@@ -77,83 +90,93 @@
+        <xsl:context-item as="element(catalog)" use="required"/>
+        <xsl:variable name="flat" as="element(flat)*" select="merge/flat[.=$true-content]"/>
+        <xsl:variable name="as-is" as="element(as-is)*" select="merge/as-is[.=$true-content]"/>
+        <xsl:variable name="custom" as="element(custom)*" select="merge/custom"/>
+        <xsl:if test="count($flat) + count($as-is) + count($custom) gt 1">
+            <xsl:call-template name="mh:message-handler">
+                <xsl:with-param name="text">Found multiple structuring directives. Choose at most one from: flat (default), as-is, custom.</xsl:with-param>
+                <xsl:with-param name="message-type">Error</xsl:with-param>
+                <xsl:with-param name="terminate" select="true()"/>
+            </xsl:call-template>
+        </xsl:if>
+    </xsl:template>
+
+    <xsl:template name="combine-back-matter" as="element(back-matter)?">
+        <xsl:context-item as="element(catalog)" use="required"/>
+        <xsl:where-populated>
             <back-matter>
                 <!-- Using combination logic on back matter elements. -->
-                <xsl:for-each-group select="current-group()" group-by="(@opr:id,@id,generate-id())[1]">
+                <xsl:for-each-group select="back-matter/* | selection/back-matter/*" group-by="(@opr:id,@uuid,generate-id())[1]">
                     <xsl:call-template name="combine-elements">
-                        <xsl:with-param name="who" select="current-group()"/>
+                        <!-- Take last one in group because of spec
+                            requirement id="req-backmatter-dupe". -->
+                        <xsl:with-param name="who" select="current-group()[last()]" as="element(resource)"/>
                     </xsl:call-template>
                 </xsl:for-each-group>
             </back-matter>
-        </xsl:for-each-group>
+        </xsl:where-populated>
     </xsl:template>
 
     <xsl:template match="selection">
         <xsl:apply-templates select="param   | .//group/param"/>
         <xsl:apply-templates select="control | .//group/control"/>
     </xsl:template>
 
-    <xsl:key name="control-by-id" match="control" use="@id"/>
-
-    <xsl:template match="node() | @*" mode="o:custom-merge">
-        <xsl:copy>
-            <xsl:apply-templates mode="#current" select="node() | @*"/>
-        </xsl:copy>
-    </xsl:template>
-
+    <!-- Process children but do not copy <custom> tags themselves. -->
     <xsl:template match="custom" mode="o:custom-merge">
         <xsl:apply-templates mode="#current"/>
     </xsl:template>
 
+    <!-- Apply the requested ordering.
+        Creating the properly combined sequence of inserted controls
+        is in downstream template. -->
     <xsl:template match="insert-controls" mode="o:custom-merge">
-        <xsl:variable name="inserted-controls">
+        <xsl:variable name="inserted-controls" as="element()*">
```

Comment:
Here, a sequence data type rather than a document node is important so xsl:perform-sort can sort the elements.